fire/SFMainMod.F90

```diff
@@ -902,37 +980,103 @@ subroutine  crown_damage ( currentSite )
+                   ! 0.01 = C from Van Wagner 1977 EQ4 for crown of base height 6m, 100% FMC, and FI 2500kW/m
+                   passive_crown_FI = (0.01_r8 * height_cbb *EDPftvarcon_inst%crown_ignite_energy(currentCohort%pft))**1.5_r8
+
+                   if (currentPatch%FI >= crown_fire_threshold) then ! 200 kW/m = threshold for crown fire potential
+                      
+                      ! Initiation of passive crown fire, EQ 14 Bessie and Johnson 1995
+                      ignite_crown = currentPatch%FI/passive_crown_FI
+                      
+                      if (ignite_crown >= 1.0_r8) then
+                         currentCohort%passive_crown_fire_flg = 1 ! passive crown fire ignited
+                         write(fates_log(),*) 'SF currentCohort%passive_crown_fire_flg, height_cbb = ', currentCohort%passive_crown_fire_flg, height_cbb  ! slevis diag
+                         ! "...in passive crown fires and high intensity surface
+                         ! fires trees can survive. Jack, red
+                         ! and white pine survive...scars used in dating fires"
+                         ! Johnson, E.A. 1992 Fire and Veg Dynamics: North American boreal forest. Cambridge Press
+                         currentCohort%fraction_crown_burned = EDPftvarcon_inst%crown_fire(currentCohort%pft)
+
+                         ! Critical intensity for active crowning (kW/m)
+                         ! EQ 12 Bessie and Johnson 1995
+                         ! Fuels / 0.45 to get biomass but note that the 0.45
+                         ! cancels out and could be removed. Also dividing
+                         ! critical_mass_flow_rate by 3.34, an empirical
+                         ! constant in Bessie & Johnson 1995
+                         leaf_c = currentCohort%prt%GetState(leaf_organ, all_carbon_elements)
+                         currentPatch%fuel_bulkd = currentCohort%n * leaf_c /  &
+                            (0.45_r8 * currentCohort%c_area * crown_depth)
+                         active_crown_FI = critical_mass_flow_rate *  &
+                            low_heat_of_combustion * currentPatch%sum_fuel /  &
+                            (0.45_r8 * 3.34_r8 * currentPatch%fuel_bulkd)
+
+                         ! Initiate active crown fire?
+                         ! EQ 14b Bessie & Johnson 1995
+                         ignite_active_crown = currentPatch%FI / active_crown_FI
+
+                         if (ignite_active_crown >= 1.0_r8) then
+                            currentCohort%active_crown_fire_flg = 1  ! active crown fire ignited
+                            write(fates_log(),*) 'SF currentCohort%active_crown_fire_flg, currentPatch%fuel_bulkd = ', currentCohort%active_crown_fire_flg, currentPatch%fuel_bulkd  ! slevis diag
+                            currentCohort%fraction_crown_burned = 1.0_r8
```

Comment:
If we get active crown fire conditions for a cohort, after cohort loop, then we call characteristics of fuel and update Patch%fuel to include crown foliage from active crown fire cohorts.

Reply:
After you wrote this, we discussed this in yesterday's code-review call, but I will also put a summary in writing here:

Instead of calling `characteristics_of_fuel` a second time, which will cause unwanted side effects, I inserted a short piece of code to subroutine `rate_of_spread` that updates only the specific characteristics of fuel needed.

@glemieux and @rgknox suggested making that insertion a separate subroutine, and I agree that it would be good to do during the clean-up phase of this work.